File: AndroidJava/JavaAndroidFirstApp/app/src/main/java/com/theannguyen/javaandroidfirstapp/MainActivity.java
package com.theannguyen.javaandroidfirstapp;

import android.app.Activity;
import android.content.Intent;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    Button button;
    TextView textView;
    final int REQUEST_CODE = 55;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //TODO
        button = (Button) findViewById(R.id.button);
        textView = (TextView) findViewById(R.id.textView);
        final Intent intent = new Intent(MainActivity.this, Main2Activity.class);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivityForResult(intent,REQUEST_CODE);
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        if(requestCode == REQUEST_CODE && resultCode == Activity.RESULT_OK){
            String name = data.getStringExtra("name");
            textView.setText("RESULT_OK _ " + name);
            button.setVisibility(View.INVISIBLE);
        }
        if(requestCode == REQUEST_CODE && resultCode == Activity.RESULT_CANCELED){
            //String name = data.getStringExtra("");
            textView.setText("RESULT_CANCELED");
            button.setVisibility(View.VISIBLE);
        }
    }

    @Override
    protected void onStart() {
        super.onStart();
    }

    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    @Override
    protected void onStop() {
        super.onStop();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
    @Override
    protected void onRestart() {
        super.onRestart();
    }
}

File: Android_Kotlin/Intent/app/src/main/java/com/theannguyen/intent/MainActivity.kt
package com.theannguyen.intent

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val hocSinh:HocSinh = HocSinh("Nguyen The An", 20)
        btnClick.setOnClickListener {
            val intent:Intent = Intent(this,Main2Activity::class.java)
            intent.putExtra("data",100)
            intent.putExtra("hocsinh",hocSinh)
            startActivity(intent)
        }

    }
}
